From 5e22a7181796a696348cb1a38e9187367d99f107 Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Fri, 18 Jan 2019 12:11:26 +0100
Subject: [PATCH 09/16] arm64: dt: fsl-imx8mq-tqma8mq: restrict reserved mem
 for CMA

module has only 1GiB memory, limit CMA size to 256 MiB

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 .../boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi    | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
index 6be8c5085f7c..079741ccd53f 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
@@ -33,6 +33,21 @@
 		regulator-max-microvolt = <3300000>;
 		regulator-always-on;
 	};
+
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		/* global autoconfigured region for contiguous allocations */
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0 0x10000000>;
+			alloc-ranges = <0 0x40000000 0 0x40000000>;
+			linux,cma-default;
+		};
+	};
 };
 
 &iomuxc {
-- 
2.17.1

